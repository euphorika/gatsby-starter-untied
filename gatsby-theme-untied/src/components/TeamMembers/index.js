import React from 'react'
import PropTypes from 'prop-types'

import styles from './styles.module.styl'

const TeamMembers = ({ headline, children }) => {
  const headlineElement = headline ? (
    <h2 className={styles.headline}>{headline}</h2>
  ) : null
  return (
    <section className={styles.teamMemberSection}>
      <div className={styles.innerContainer}>
        <div>{headlineElement}</div>
        {children}
      </div>
    </section>
  )
}

TeamMembers.propTypes = {
  children: PropTypes.node.isRequired,
  headline: PropTypes.string,
}

export default TeamMembers
