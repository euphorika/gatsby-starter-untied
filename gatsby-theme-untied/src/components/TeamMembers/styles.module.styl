@require '../../themes/standard/config/breakpoints'
@require '../../themes/standard/config/colors'
@require '../../themes/standard/global/container'

.team-member-section
  margin 0

  .inner-container
    display flex
    justify-content space-between
