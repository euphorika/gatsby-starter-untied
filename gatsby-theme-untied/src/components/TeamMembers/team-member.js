/** @jsx jsx */
import { jsx } from 'theme-ui'
import PropTypes from 'prop-types'
import Img from 'gatsby-image'

const TeamMember = ({ children, name, position, imgFixed }) => (
  <div
    sx={{
      float: ['none', 'left'],
      width: ['auto', '50%'],
      mt: [4, 0],
      px: 2,
      py: 3,
    }}
  >
    <div sx={{ display: 'flex' }}>
      <Img fixed={imgFixed} />
      <div
        sx={{
          display: 'flex',
          flexDirection: 'column',
          justifyContent: 'center',
          pl: 2,
        }}
      >
        <h3
          sx={{
            fontSize: 3,
            fontWeight: 'heading',
            mb: 1,
          }}
        >
          {name}
        </h3>
        <p
          sx={{
            fontSize: 2,
            margin: 0,
          }}
        >
          {position}
        </p>
      </div>
    </div>
    <div
      sx={{
        fontSize: 1,
        lineHeight: '26px',
        mt: 3,
        mr: [0, 2],
      }}
    >
      {children}
    </div>
  </div>
)

TeamMember.propTypes = {
  children: PropTypes.node.isRequired,
  name: PropTypes.string.isRequired,
}

export default TeamMember
