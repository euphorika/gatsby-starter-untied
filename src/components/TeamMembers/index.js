import React from 'react'

import styles from './styles.module.styl'

const TeamMembers = () => (
  <section className={styles.TeamMemberSection}>Team Members</section>
)

export default TeamMembers
