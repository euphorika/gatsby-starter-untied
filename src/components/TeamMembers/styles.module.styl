@require '../../themes/standard/config/breakpoints'
@require '../../themes/standard/config/colors'

.team-member-section
  margin 0
