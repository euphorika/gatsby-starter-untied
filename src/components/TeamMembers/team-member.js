import React from 'react'
import PropTypes from 'prop-types'
import Img from 'gatsby-image'

const TeamMember = ({ children, name, position, imgFixed }) => (
  <div className="team-member">
    <Img fixed={imgFixed} />
    <h3>{name}</h3>
    <p>
      <strong>{position}</strong>
    </p>
    <div className="bio">{children}</div>
  </div>
)

TeamMember.propTypes = {
  children: PropTypes.node.isRequired,
  name: PropTypes.string.isRequired,
}

export default TeamMember
