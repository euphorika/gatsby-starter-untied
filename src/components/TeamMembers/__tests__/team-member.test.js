import React from 'React'
import renderer from 'react-test-renderer'
import TeamMembers from '../'

describe('TeamMember', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<TeamMembers />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
