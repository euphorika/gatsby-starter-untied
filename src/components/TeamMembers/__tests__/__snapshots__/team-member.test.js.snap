// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamMember renders correctly 1`] = `
<section
  className="TeamMemberSection"
>
  Team Member
</section>
`;
